datasource/utils.go

```diff
@@ -4,6 +4,49 @@ package datasource
 
 import "strconv"
 
+// Any crseAttrValue's that don't fall into these are labeled "None"
+var divisionalAreas = [3]string{"A&H", "SCI", "SOC"}
+
+func getGenEdInfo(intAttributes []any) []string {
+
+	// Get the gen-ed areas from the course attributes.
+	areas := []string{}
+	for _, item := range intAttributes {
+		attrMap := item.(map[string]any)
+		if attrMap["crseAttribute"] == "LGE" {
+			areas = append(areas, attrMap["crseAttrValue"].(string))
+		}
+	}
+	return areas // if empty, not a GENED.
+}
+
+func checkDivisionalArea(val string) bool {
+	for _, area := range divisionalAreas {
+		if val == area {
+			return true
+		}
+	}
+	return false
+}
+
+func getDivisionalInfo(intAttributes []interface{}) []string {
+
+	// Type conversion, since we're dealing with an interface.
+	areas := []string{}
+	for _, item := range intAttributes {
+		attrMap, _ := item.(map[string]any)
```

Comment:
Did you mean to ignore a failed type assertion?
```suggestion
		attrMap := item.(map[string]any)
```